### DISCUSSION TITLE
Portfolio analytics and management functions

### DISCUSSION THREAD

**Discussion by peeeffchang** [mbk-dev/okama#16] [Ideas]
Thanks for putting the work in this project. Is there any plan to expand to other areas for portfolio management? E.g.
- Factor decomposition of portfolio
- Risk attribution
- Optimization with respect to more general criteria (such as zero exposure in a certain factor)
I would be interested to contribute as well.

**Comment by chilango74** [comment]
@peeeffchang We have some aspects of mentioned in the [Roadmap](https://github.com/mbk-dev/okama#roadmap).

> Add different utility functions for optimizers: semi-deviation, VaR, CVaR, drawdowns etc.

I'd like to work on more general framework for utility functions in the optimizer. Even now we are using "black box" optimization for different "non classical" functions like geometric mean. 

In general we use CFA time series quantative analysis suggestions as a guideline of future development. Factor decomposition and risk attribution go well with it.

And ofcourse all kind of contribution are quite wellcome.
